### ISSUE TITLE
cannot locate symbol sk_num

### ISSUE THREAD

**Issue by ufobat** [sergot/openssl#68]
```
martin@sinclair:~/repos$ zef install OpenSSL
===> Searching for: OpenSSL
===> Testing: OpenSSL:ver<0.1.21>:auth<github:sergot>
# NETWORK_TESTING was not set
Cannot locate symbol 'sk_num' in native library 'libssl.so'
  in method setup at /home/martin/.rakudobrew/moar-2018.12/install/share/perl6/sources/947BDAB9F96E0E5FCCB383124F923A6BF6F8D76B (NativeCall) line 283
  in method CALL-ME at /home/martin/.rakudobrew/moar-2018.12/install/share/perl6/sources/947BDAB9F96E0E5FCCB383124F923A6BF6F8D76B (NativeCall) line 570
  in block <unit> at t/10-client-ca-file.t line 10

# Looks like you planned 7 tests, but ran 1
# NETWORK_TESTING was not set
===> Testing [FAIL]: OpenSSL:ver<0.1.21>:auth<github:sergot>
Aborting due to test failure: OpenSSL:ver<0.1.21>:auth<github:sergot> (use --force-test to override)
martin@sinclair:~/repos$

```

**Comment by vendethiel** [comment]
has anyone investigated this issue?

**Comment by Xliff** [comment]
Yes, this is a repeat of jnthn/p6-io-socket-async-ssl/#34. Unfortunately, this is going to become more common, since libssl1.0 has now been removed from the latest KUbuntu. Here's the workaround:

- Download libssl1.0 from here[https://ubuntu.pkgs.org/18.04/ubuntu-updates-main-amd64/libssl1.0.0_1.0.2n-1ubuntu5.3_amd64.deb.htmlhttps://ubuntu.pkgs.org/18.04/ubuntu-updates-main-amd64/libssl1.0.0_1.0.2n-1ubuntu5.3_amd64.deb.html]
- Install it via `sudo dpkg -i  libssl1.0.0_1.0.2n-1ubuntu5.3_amd64.deb`
- Download libssl1.0-dev from here[http://ftp.br.debian.org/debian/pool/main/o/openssl1.0/libssl1.0-dev_1.0.2r-1~deb9u1_amd64.deb]
- Install if via `sudo dpkg -i --force-all http://ftp.br.debian.org/debian/pool/main/o/openssl1.0/libssl1.0-dev_1.0.2r-1~deb9u1_amd64.deb`
- You will then have to handle the symbolic link, manually: `cd /usr/lib/x86_64-linux-gnu; sudo ln -sf libssl.so.1.0.0 libssl.so`

You should now be able to install this module without errors.

**Comment by rbt** [comment]
Indeed. The libssl1.0-dev package has even been removed from Ubuntu 19.10. Seems safer to require OpenSSL 1.1 than 1.0 at this point.

**Comment by Xliff** [comment]
Is @sergot active?

**Comment by sergot** [comment]
Unfortunately I'm not that active anymore @Xliff , will try to find some time this week.

**Comment by Altai-man** [comment]
Just saw this on fedora 32, OpenSSL 1.1.1g FIPS, BUT...

If I clone the repo and `zef install .`, it installs just fine. If I do `zef install OpenSSL` without a local copy, it fails with sk_num not being found, which is rather odd.

**Comment by usev6** [comment]
I think this problem should no longer happen. At least I was able to install OpenSSL on Debian 10 and Ubuntu 20.04.2 LTS (both coming with OpenSSL 1.1.1).

If I'm not mistaken, the following changes made things work:
* https://github.com/sergot/openssl/pull/65
* the work from @ugexe for https://github.com/sergot/openssl/issues/82 (which removed stale precomp files)

Please note: The second change was only necessary if the libssl.so was not already present when attempting to install OpenSSL the first time.

However, I think (hope) that this issue could be closed now. It would be great if someone else double checks.
